### PR TITLE
Support enum to string comparision

### PR DESCRIPTION
**Pull request by bhaarathms**
### Description

Support string to Enum conversion in filters

### Checklist (Uncheck if it is not completed)

- [ X] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

No

### CODE REVIEW

**Comment by bhaarathms** [comment]
This PR is same as https://github.com/OData/odata.net/pull/968 
I messed something up while rebasing with master on the branch corresponding old PR. So I created this one afresh. 
I will close the old PR.

File: test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs [hunk 1]
```diff
@@ -139,33 +139,19 @@ public override void PromoteBinaryOperandTypes(BinaryOperatorKind binaryOperator
         [Fact]
         public void TestStringAsEnumInFunctionParameter()
         {
-            this.TestStringAsEnum(
```
Comment (robward-ms) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs]: Why is this test being removed?

Comment (bhaarathms) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/ODataUriResolverTests.cs]: With the change in this PR String to Enum Conversion is supported. The two tests here check for error messages on converting a string to Enum. So they are no longer valid. So I am removing them and adding a test to verify the conversion.

File: src/Microsoft.OData.Core/UriParser/Binders/MetadataBindingUtils.cs [hunk 2]
```diff
@@ -55,14 +55,18 @@ internal static SingleValueNode ConvertToTypeIfNeeded(SingleValueNode source, IE
                     return new ConvertNode(source, targetTypeReference);
                 }
 
+                ConstantNode constantNode = source as ConstantNode;
+                if (source.TypeReference.IsString() && targetTypeReference.IsEnum())
+                {
+                    return new ConstantNode(new ODataEnumValue(constantNode.Value.ToString(), targetTypeReference.Definition.ToString()), constantNode.Value.ToString(), targetTypeReference);
```
Comment (mtroth-microsoft) [inline src/Microsoft.OData.Core/UriParser/Binders/MetadataBindingUtils.cs]: >constantNode [](start = 63, length = 12)

how do you know this is not null?

Comment (bhaarathms) [reply src/Microsoft.OData.Core/UriParser/Binders/MetadataBindingUtils.cs]: added null checks

**Pull request merged**